app/src/main/java/com/nutomic/syncthingandroid/activities/FolderActivity.java

```diff
@@ -431,15 +431,7 @@ private void initFolder() {
                 ? getIntent().getStringExtra(EXTRA_FOLDER_ID)
                 : generateRandomFolderId();
         mFolder.label = getIntent().getStringExtra(EXTRA_FOLDER_LABEL);
-        if (Build.VERSION.SDK_INT != Build.VERSION_CODES.M) {
-            // Scan every hour (in case real time change detection failed)
-            mFolder.rescanIntervalS = 3600;
-        }
-        else {
-            // FileObserver is broken on Marshmallow.
-            // https://github.com/syncthing/syncthing-android/issues/787
-            mFolder.rescanIntervalS = 300;
-        }
+        mFolder.rescanIntervalS = 3600;
```

Comment:
Add a comment explaining why we are setting this. Also, shouldn't we set this to 60s if the watcher is disabled?

Reply:
Well the comment is still a lie, as default is 60 and suggested when watcher is enabled is 3600

Reply:
This sets the default values for folders created in the app, right? Where is the fs watcher enabled? I think the interval should be set at the same place.

Reply:
It's always enabled and this sets it for all folders unconditionally

Reply:
@imsodin: Yes, only folders created in the app get this rescanInterval as default. see FolderActivity.java line 155.